### DISCUSSION TITLE
Combobox - How to prevent active item reset when query changes if it's still in the list?

### DISCUSSION THREAD

**Discussion by ItaiYosephi** [ariakit/ariakit#3936] [Questions] [answered]
in the following video taken from the Command Menu example,  is there a way to keep`Import Extension` item active?
https://github.com/ariakit/ariakit/assets/40234776/f745ac08-32ce-4f46-ba3c-9498beaee328

**Comment by diegohaz** [comment, marked as answer]
I haven't tried it, but I think you could use [`getAutoSelectId`](https://ariakit.org/reference/combobox#getautoselectid) and return the store's activeId.

Edit: Yes, it looks like this works:

```tsx
  const combobox = Ariakit.useComboboxContext()!;

  <Ariakit.Combobox
    autoSelect="always"
    getAutoSelectId={(items) => {
      const { activeId } = combobox.getState();
      if (!activeId) return;
      const activeItem = items.find((item) => item.id === activeId);
      return activeItem?.id;
    }}
  />
```
